src/libraries/System.Private.CoreLib/src/System/Numerics/BitOperations.cs

```diff
@@ -243,6 +244,19 @@ public static int PopCount(uint value)
                 return (int)Popcnt.PopCount(value);
             }
 
+            if (AdvSimd.Arm64.IsSupported)
+            {
+                // PopCount works on vector so convert input value to vector first.
+
+                // Vector64.CreateScalar(uint) generates suboptimal code by storing and
+                // loading the result to memory.
+                // See https://github.com/dotnet/runtime/issues/35976 for details.
+                // Hence use Vector4.Create(ulong) to create Vector64<ulong> and operate on that.
+                Vector64<ulong> input = Vector64.Create((ulong)value);
+                Vector64<byte> aggregated = AdvSimd.Arm64.AddAcross(AdvSimd.PopCount(input.AsByte()));
+                return AdvSimd.Extract(aggregated, 0);
```

Comment:
The following might be more readable and we should be optimizing it to the same thing (I'm not sure if we've enabled this for ARM64 yet though)
```suggestion
                return aggregated.ToScalar();
```

Reply:
That's correct. `.ToScalar()` is ongoing work in https://github.com/dotnet/runtime/pull/35736